### PR TITLE
Use sets.String replace sets.Set[string]

### PR DESCRIPTION
**Pull request by sword-jin**
Only internal changes:

- replace all sets.Set[string] by sets.String 
- add String() function to Set

### CODE REVIEW

**Review by hzxuzhonghu — APPROVED** [review]

File: cni/pkg/plugin/plugin.go [hunk 1]
```diff
@@ -278,7 +278,7 @@ func doRun(args *skel.CmdArgs, conf *Config) error {
 	}
 
 	if !pi.Containers.Contains(ISTIOPROXY) {
-		log.Infof("excluded because it does not have istio-proxy container (have %v)", sets.SortedList(pi.Containers))
+		log.Infof("excluded because it does not have istio-proxy container (have %s)", sets.SortedList(pi.Containers))
```
Comment (howardjohn) [inline cni/pkg/plugin/plugin.go:281]: a slice is not `%s` compatible?

File: pkg/util/sets/set_test.go [hunk 2]
```diff
@@ -338,3 +338,18 @@ func TestMapOfSet(t *testing.T) {
...
+	set := New(elements...)
+
+	// in stringSet, we are pretty sure there will be three elements
+	stringSet := String{}
+	for i := 0; i < 1000; i++ {
+		stringSet.Insert(set.String())
+		if stringSet.Len() == 3 {
+			break
+		}
+	}
```
Comment (howardjohn) [inline pkg/util/sets/set_test.go:353]: This feels a bit excessive to test this like this...

**Review by howardjohn — CHANGES_REQUESTED** [review]

File: cni/pkg/plugin/plugin.go (hunk 1 above)
Comment (sword-jin) [reply cni/pkg/plugin/plugin.go:281]: You are right, this is a []T not Set[T], fixed

File: pkg/util/sets/set_test.go (hunk 2 above)
Comment (sword-jin) [reply pkg/util/sets/set_test.go]: Simpilified

**Review by howardjohn — APPROVED** [review]

File: pkg/util/sets/set.go [hunk 3]
```diff
@@ -233,6 +235,13 @@ func (s Set[T]) IsEmpty() bool {
 	return len(s) == 0
 }
 
+// String returns a string representation of the set.
+// Be aware that the order of elements is random so the string representation may vary.
+// Use it only for debugging and logging.
+func (s Set[T]) String() string {
+	return fmt.Sprintf("%v", s.UnsortedList())
```
Comment (wulianglongrd) [inline pkg/util/sets/set.go:242]: Would it look better using `SortedList`?

**Pull request merged**